packages/core/src/common/registry/resolve-id-registry.ts

```diff
@@ -48,6 +55,14 @@ export default class ResolveIdRegistry<TType, TToken extends { [key: string]: un
             results.push(result);
         });
 
+        const uniqueMatches = [...Array.from(new Set(results.map((result) => result.matches)))];
+
+        if (results.length > 1 && uniqueMatches.length === 1) {
```

Comment:
`results` always contains all registeredTokens, hence it should be always greater than 1. See L55.
I think you want to do the same check on the `matched`, L66.

Reply:
On a second thought, I would find the biggest matches value and how many tokens have that value.
If the number is greater than one, throw an error then.
My concern is that `uniqueMatches.length === 1` might not cover all equal matches cases.

Reply:
Yes I think I am checking for the case where there are more than one `registeredTokens` in result. 

I am looking for testing the case where there are multiple tokens but for `uniqueMatches` only have a length of 1. 

Right now I think it works well for the existing test cases. Do we want to test for if `uniqueMatches > 1`?

Reply:
`results.length` equals `registeredTokens.length`.

Is it possible that we have a result like this?
```
[{ matches: 2 }, { matches: 2 }, { matches: 1 }, { matches: 1 }]
```

Reply:
I'd move the check down after the sort. i.e.:

```ts
const matchedResults = results
    .sort((a, b) => b.matches - a.matches)
    .filter((result) => result.matches > 0);

if (matchedResults.length > 1 && matchedResults[0].matches === matchedResults[1].matches) {
    if (getEnvironment() === 'development') {
        getDefaultLogger().warn('The provided query matches at least two strategies with the same specificity. This warning can be resolved by making their resolve ID more specific.');
    }
}

const matched = matchedResults[0];

if (matched && matched.token) {
    return matched.token;
}
```

Also, I'd make it warning in development rather than an error.

Reply:
Thanks @davidchin that seems to be good, but I am still on the fence regarding the error vs warn. Since warnings can be ignored and can result in issues if pushed to prod. Thoughts @bc-peng?

Reply:
Thank you both.
We may want to eliminate uncertainty in production, so an error seems more appropriate to me.

Reply:
I suggested warning because it's not really an error if it resolves to the right strategy even if there are more than one match. But if you think error is more appropriate, I'm also happy with it. 🙂  Just make sure you do manual testing for payment methods that don't have e2e tests.

Reply:
Thanks for the input, I will update and do the necessary testing 😄